spec/services/allocation_updater_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.describe AllocationUpdater do
+  let(:mock_request) { instance_double(Computacenter::OutgoingAPI::CapUpdateRequest, timestamp: Time.zone.now, payload_id: '123456789') }
+
+  subject(:service) do
+    described_class.new(school: school, device_type: 'std_device', value: 100)
+  end
+
+  context 'allocation exists and school can order full allocation' do
+    let(:school) { create(:school, :in_lockdown) }
+    let(:update_service) { instance_double('SchoolOrderStateAndCapUpdateService') }
+
+    before do
+      create(:school_device_allocation,
+             :with_std_allocation,
+             allocation: 10,
+             cap: 10,
+             school: school)
+
+      allow(Computacenter::OutgoingAPI::CapUpdateRequest).to receive(:new).and_return(mock_request)
+      allow(mock_request).to receive(:post!)
+    end
+
+    it 'updates cap to match allocation' do
+      service.call
+      allocation = school.std_device_allocation.reload
+      expect(allocation.allocation).to be(100)
+      expect(allocation.cap).to be(100)
+    end
+
+    it 'calls notifies computacenter with cap updates' do
+      expect(SchoolOrderStateAndCapUpdateService).to receive(:new).with(school: school, order_state: school.order_state, std_device_cap: 100, coms_device_cap: 0).and_call_original
+
+      service.call
+
+      expect(mock_request).to have_received(:post!).twice
```

Comment:
I don't think this expectation belongs in this spec - as far as testing this service is concerned, it shouldn't need to know about the internals of what the `SchoolOrderStateAndCapUpdateService` does when you call `.update!`, just that an instance gets created with the right params and `.update!` gets called on it. That can be tested with a mock/double, which should mean you don't need to disable the RSpec MessageSpies rubocop option or mock the HTTP request.